.github/workflows/ci.yml

```diff
@@ -13,27 +13,46 @@ jobs:
         name: [ "Linux x64 (Ubuntu 20.04) - GCC 10" ]
         os: [ ubuntu-20.04 ]
         sanitize: [ false ]
-        build:
-          - { cc: gcc, cxx: g++, linker: ld, shell: bash }
+        build: { cc: gcc, cxx: g++, linker: ld }
+        shell: bash
+        prefix: "/usr/local"
+        sudo: "sudo"
+
         include:
           - name: "Linux x64 (Ubuntu 20.04) - Clang 10 with ASan and UBSan"
             os: ubuntu-20.04
             sanitize: true
-            build: { cc: clang-10, cxx: clang++-10, linker: ld.lld-10, shell: bash }
+            build: { cc: clang-10, cxx: clang++-10, linker: ld.lld-10 }
+            shell: bash
+            prefix: "/usr/local"
+            sudo: "sudo"
+
           - name: "macOS (10.15) - Xcode 12.3"
             os: macos-10.15
             sanitize: false
-            build: { cc: clang, cxx: clang++, linker: ld.lld, shell: bash }
+            build: { cc: clang, cxx: clang++, linker: ld.lld }
+            shell: bash
+            prefix: "/usr/local"
+            sudo: "sudo"
+
           - name: "macOS (10.15) - Xcode 12.3 - M1"
             os: macos-m1
             sanitize: false
-            build: { cc: clang, cxx: clang++, linker: ld.lld, shell: "/usr/bin/arch -arch arm64e /bin/bash -l {0}" }
+            build: { cc: clang, cxx: clang++, linker: ld.lld }
+            shell: "/usr/bin/arch -arch arm64e /bin/bash -l {0}"
+            prefix: "/Users/Administrator/vips"
+            sudo: ""
 
     defaults:
       run:
         shell: ${{ matrix.build.shell }} 
 
     steps:
+      - name: Set prefix
+        env:
+          PREFIX: ${{ matrix.prefix }}
+          SUDO: ${{ matrix.sudo }}
+
```

Comment:
```suggestion
```

Reply:
(it was failing with "every step must define a `uses` or `run` key")

Reply:
Oh of course. I saw the error, but there was no line number. I've moved the env to the job rather than the step.

Reply:
Ah yes, that's much neater. Perhaps we can move this step to there as well?
https://github.com/libvips/libvips/blob/ea2ad5ffb03871a7747bb4ff3bf447c6bc32bf3a/.github/workflows/ci.yml#L103-L108